stress/shell/src/main/java/alluxio/stress/cli/MaxFileBench.java

```diff
@@ -0,0 +1,224 @@
+    ExecutorService service = ExecutorServiceFactories.fixedThreadPool("maxfile-bench-thread",
+        mParameters.mThreads).create();
+
+    List<Callable<Void>> callables = new ArrayList<>(mParameters.mThreads);
+    for (int i = 0; i < mParameters.mThreads; i++) {
+      callables.add(new AlluxioNativeMaxFileThread(i, mCachedNativeFs[i % mCachedNativeFs.length]));
+    }
+    LOG.info("Starting {} bench threads", callables.size());
+    long startMs = CommonUtils.getCurrentMs();
+    service.invokeAll(callables, FormatUtils.parseTimeSize(mBaseParameters.mBenchTimeout),
+        TimeUnit.MILLISECONDS);
+    mTotalResults.setDurationMs(CommonUtils.getCurrentMs() - startMs);
+    LOG.info("Bench threads finished");
+
+    service.shutdownNow();
+    service.awaitTermination(30, TimeUnit.SECONDS);
+
+    return mTotalResults;
+  }
+
+  private final class AlluxioNativeMaxFileThread implements Callable<Void> {
+    private static final int INIT_WAIT_TIME_MS = 100;
+    private static final int MAX_WAIT_TIME_MS = 15_000;
+    private static final int OPERATION_TIMEOUT_MS = 2_000;
+
+    private int mWaitTimeMs = INIT_WAIT_TIME_MS;
+    private final ExecutorService mExecutor = Executors.newSingleThreadExecutor();
+    private final MasterBenchTaskResult mResult = new MasterBenchTaskResult();
+
+    private final int mId;
+    private final FileSystem mFs;
+    private final Path mBasePath;
+    private final Path mFixedBasePath;
+
+    AlluxioNativeMaxFileThread(int id, FileSystem fs) {
+      mId = id;
+      mFs = fs;
+      mBasePath =
+          new Path(PathUtils.concatPath(mParameters.mBasePath, "files", mId));
+      mFixedBasePath = new Path(mBasePath, "fixed");
+      LOG.info("[{}]: basePath: {}, fixedBasePath: {}", mId, mBasePath, mFixedBasePath);
+    }
+
+    @Override
+    public Void call() throws Exception {
+      mResult.setRecordStartMs(CommonUtils.getCurrentMs());
+      AtomicLong localCounter = new AtomicLong();
+      while (!sFinish.get()) {
+        if (Thread.currentThread().isInterrupted()) {
+          break;
+        }
+        long increment = localCounter.getAndIncrement();
+        if (increment % 100_000 == 0) {
+          LOG.info("[{}] Created {} files", mId, increment);
+        }
+        try {
+          mExecutor.submit(() -> {
+            try {
+              applyOperation(increment);
+            } catch (Exception e) {
+              throw new RuntimeException(e);
+            }
+          }).get(OPERATION_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+        } catch (Exception e) {
+          if (mWaitTimeMs > MAX_WAIT_TIME_MS) {
+            LOG.info("[{}] Waiting timeout, ending: {}", mId, e);
+            sFinish.set(true);
+            break;
+          } else {
+            LOG.info("[{}] Failed, sleeping for {}: {}", mId, mWaitTimeMs, e);
+            Thread.sleep(mWaitTimeMs);
+            // exponential backoff retry
+            mWaitTimeMs *= 2;
```

Comment:
Can't we use `RetryUtils` instead?

Reply:
`ExponentialBackoffRetry` has a failure condition based on # attempts, not on how long the timeout is.
Also there is no way of resetting the `RetryPolicy` other than creating a new `RetryPolicy`.

Reply:
How about `ExponentialTimeBoundedRetry`?
It's ok to recreate the policy. It's better to avoid doing custom logic here.

Reply:
I moved to `ExponentialBackoffRetry`.